src/accelerate/utils/fsdp_utils.py

```diff
@@ -58,8 +46,12 @@ def _set_model_state_dict(model, state_dict, adapter_only=False):
 
 
 def save_fsdp_model(fsdp_plugin, accelerator, model, output_dir, model_index=0, adapter_only=False):
-    os.makedirs(output_dir, exist_ok=True)
+    import torch.distributed.checkpoint as dist_cp
+    from torch.distributed.checkpoint.default_planner import DefaultSavePlanner
+    from torch.distributed.fsdp.fully_sharded_data_parallel import FullyShardedDataParallel as FSDP
+    from torch.distributed.fsdp.fully_sharded_data_parallel import StateDictType
```

Comment:
this feels somewhat yucky - since it's going to run the imports many times over - in particular because you replicate this exact import code multiple times below.

Why not move these imports into its own file and call them imports only if they haven't been loaded yet?

regardless it should be safe to assume that if a user configured accelerate to use FSDP it'll need all these save/load calls - so from the performance point of view all of them can be moved into a single place and loaded once - if the user asked for FSDP.

Reply:
Will need to think of an ideal folder structure, but I'll start with some small version of this in this PR, since we're touching those imports now (specifically sectioning out to a different file)

Reply:
Also the difference with megatron is we check for a package being installed. So while we can do similar for deepspeed, we cannot do this for fsdp (as it doesn't actually check for configurations, only if a package is available in the env)

Reply:
but it should be enough to check whether FSDP is being configured to be used - configured? import, not configured? don't import

Reply:
the dependency chain could be:

this_component_is_needed_to_continue -> do we have it installed -> if so import the needed bits.

i.e. the logic that decides to whether import something or not is driven by the configuration of accelerate and not by availability of a 3rd party package.

Reply:
Hmm, but imports are cached in Python, so the 2nd time this is called, they don't add any overhead. Or am I missing something?

Reply:
sure, but this just feels weird. 

imports don't belong to methods/functions and surely not when you repeat the same imports multiple times in multiple functions. Nothing stops you from doing that, of course.

imports are sort of a singleton thing and certainly we occasionally see an odd import in some function, but it's usually an exception and this happens more often in a throw-away code.